libvast/include/vast/detail/settings.hpp

```diff
@@ -32,6 +32,16 @@ void merge_settings(const caf::settings& src, caf::settings& dst,
 caf::expected<uint64_t>
 get_bytesize(caf::settings opts, std::string_view key, uint64_t defval);
 
+/// @brief CAF 0.18 no longer supports comma separated values while parsing list
+/// configuration options. This function converts a comma separated list
+/// argument into one that is acceptable by CAF.
+/// @param comma_separated_list_arg command line argument with comma separated
+/// values (e.g. --plugins=web,pcap)
+/// @return command line argument which will be parsed as a list config_value by
+/// CAF. Returns empty for malformed input (e.g. lack of = in the input)
+std::string
+convert_to_caf_compatible_list_arg(const std::string& comma_separated_list_arg);
```

Comment:
I am not really happy with the use of this function.
What i tried to achieve is to create a wrapper for caf::config_option_set that knows how to handle the list values in the same manner as this function. I would replace the usage of caf::config_option_set with another class and no one would need to use this free function again.
By doing .add<std::vector> and then .parse() it would all be handled correctly.

With current approach it is needed to call this free function whenever we use list config values
Unfortunately the proposed change breaks the plugin API interface so that would require additional work and woould be a breaking change for users that wrote their own plugins. I think it is worth having this as a separate story.

Reply:
If I understood correctly then you already created this story, so we can probably resolve this. For now I think this is fine.